server/services/v1/database/secondary_index_measure.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2022-2023 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package database
+
+import (
+	"context"
+
+	"github.com/tigrisdata/tigris/internal"
+	"github.com/tigrisdata/tigris/keys"
+	"github.com/tigrisdata/tigris/query/filter"
+	"github.com/tigrisdata/tigris/schema"
+	"github.com/tigrisdata/tigris/server/config"
+	"github.com/tigrisdata/tigris/server/metrics"
+	"github.com/tigrisdata/tigris/server/transaction"
+)
+
+type secondaryIndexerWithMetrics struct {
+	q *SecondaryIndexerImpl
+}
+
+func NewSecondaryIndexer(coll *schema.DefaultCollection) SecondaryIndexer {
+	if config.DefaultConfig.Metrics.SecondaryIndex.Enabled {
+		return NewSecondaryIndexerWithMetrics(coll)
+	}
+
+	return newSecondaryIndexerImpl(coll)
+}
+
+func NewSecondaryIndexerWithMetrics(coll *schema.DefaultCollection) SecondaryIndexer {
+	q := newSecondaryIndexerImpl(coll)
+	return &secondaryIndexerWithMetrics{
+		q,
+	}
+}
+
+func (m *secondaryIndexerWithMetrics) measure(ctx context.Context, name string, f func(ctx context.Context) error) {
+	// Low level measurement wrapper that is called by the measure functions on the appropriate receiver
+	measurement := metrics.NewMeasurement("tigris.secondaryindexer", name, metrics.SecondaryIndexSpanType, metrics.GetSecondaryIndexTags(name))
```

Comment:
Put the span name in a constant where others are, so it's easy to pick a unique name.